atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -1256,9 +1256,12 @@ public PersistedSnapshot getCurrentSnapshot() {
     return currentSnapshot;
   }
 
-  public void setCurrentSnapshot(final PersistedSnapshot snapshot) {
+  public void updateCurrentSnapshot() {
     checkThread();
-    currentSnapshot = snapshot;
+    // Get the latest snapshot from snapshot store because it might have been updated already before
+    // this listener is executed
+    currentSnapshot = persistedSnapshotStore.getLatestSnapshot().orElse(null);
+    log.trace("Set currentSnapshot to {}", currentSnapshot);
     logCompactor.compactFromSnapshots(persistedSnapshotStore);
   }
```

Comment:
❓ Wdyt this can we test this behavior or do you think it is enough with the RaftRandomized? I guess it is not easy to verify.

Reply:
It is not easy to simulate the scenario outside the randomized tests. I would say Randomized tests are enough. It was quite easy to reproduce this failure.